### PR TITLE
Add basic api routes for login and register

### PR DESCRIPTION
**Pull request by utkarsh-dixit**
- [x] Added Login and Register routes
- [x] Added a logger utility using chalk to indicate log level
- [x] Create a request logger to log incoming requests to server
- [x] Update gitignore to exclude idea and dist folder for server.

### CODE REVIEW

File: server/api/controllers/auth.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  public login(req: Request, res: Response) {
+    const { email, password } = req.body;
+
+    return AuthService.loginUser(email, password)
+      .then((userToken) => {
+        return res.send({ status: "LOGGED_IN", token: userToken });
+      })
+      .catch((err) => {
+        return res.send({ status: "FAILED_TO_LOG_IN", reason: err.message });
```
Comment (flozender) [inline server/api/controllers/auth.ts]: I'd prefer if you used ```message``` instead of ```reason```

**Review by flozender — CHANGES_REQUESTED** [review]

File: server/api/controllers/auth.ts (hunk 1 above)
Comment (pingrishabh) [reply server/api/controllers/auth.ts]: +1

**Review by flozender — APPROVED** [review]
LGTM

File: server/api/controllers/auth.ts [hunk 2]
```diff
@@ -1,9 +1,34 @@
...
       message: "Welcome to Onboardio Authentication Gateway!",
     });
   }
+
+  public login(req: Request, res: Response) {
+    const { email, password } = req.body;
+
+    return AuthService.loginUser(email, password)
+      .then((userToken) => {
+        return res.send({ status: "LOGGED_IN", message: userToken });
```
Comment (flozender) [inline server/api/controllers/auth.ts]: Shouldn't this be ```token``` and not ```message```?

File: server/api/controllers/auth.ts [hunk 3]
```diff
@@ -1,9 +1,34 @@
...
+
+  public register(req: Request, res: Response) {
+    const { email, password } = req.body;
+
+    return AuthService.registerUser(email, password)
+      .then((userToken) => {
+        return res.send({ status: "USER_REGISTERED", token: userToken });
+      })
+      .catch((err) => {
+        return res.send({ status: "FAILED_TO_REGISTER", reason: err.message });
```
Comment (flozender) [inline server/api/controllers/auth.ts]: please change this as well!

**Review by flozender — CHANGES_REQUESTED** [review]
Slight changes before we can merge, please check the comments

File: server/api/controllers/auth.ts (hunk 3 above)
Comment (utkarsh-dixit) [reply server/api/controllers/auth.ts]: I don't know how this didn't get committed. Resolved now

File: server/api/controllers/auth.ts (hunk 2 above)
Comment (utkarsh-dixit) [reply server/api/controllers/auth.ts]: Fixed it now

**Review by flozender — APPROVED** [review]
LGTM

File: server/api/controllers/auth.ts [hunk 4]
```diff
@@ -1,9 +1,34 @@
 import { Request, Response } from "express";
+import { AuthService } from "../services/authService";
```
Comment (pingrishabh) [inline server/api/controllers/auth.ts:3]: +1 for using services. Would keep the controller tidy!

**Review by pingrishabh — APPROVED** [review]
LGTM! Great stuff.

**Pull request merged**